src/test/ui/variance/variance-associated-types2.rs

```diff
@@ -0,0 +1,14 @@
+trait Foo {
```

Comment:
Can we add a comment here? I would say something like

// Test that `dyn Foo<Bar = T>` is invariant with respect to `T`.
// Failure to enforce invariance here can be weaponized, see #12345 for details.

Only with the correct issue number, of course.